app/lib/katello/validators/kickstart_repository_validator.rb

```diff
@@ -0,0 +1,29 @@
+module Katello
+  module Validators
+    class KickstartRepositoryValidator < ActiveModel::Validator
+      def validate(record)
+        ks_repo_id = kickstart_repository_id(record)
+        return unless ks_repo_id
+
+        msg = if !record.operatingsystem.is_a?(Redhat)
+                _("Kickstart repositories can only be associated with Red Hat hosts")
+              elsif record.content_source.blank?
+                _("Please select a content source before assigning a kickstart repository")
+              elsif record.operatingsystem.kickstart_repos(record).none? { |repo| repo[:id] == ks_repo_id }
+                _("No such repository in the assigned content view, environment, content source, OS, and architecture")
+              end
```

Comment:
I also recommend adding a check for the presence of OS and arch because we cant set the kickstart_repository_id without those.